app/grandchallenge/profiles/models.py

```diff
@@ -146,20 +166,58 @@ def get_unsubscribe_link(self, *, subscription_type):
             else:
                 raise ValueError("User has opted out of newsletter emails")
         elif subscription_type == EmailSubscriptionTypes.NOTIFICATION:
-            if self.receive_notification_emails:
+            if (
+                self.notification_email_choice
+                == NotificationEmailOptions.DISABLED
+            ):
+                raise ValueError("User has opted out of notification emails")
+            else:
                 return reverse(
                     "notification-unsubscribe",
                     kwargs={"token": self.unsubscribe_token},
                 )
-            else:
-                raise ValueError("User has opted out of notification emails")
         elif subscription_type == EmailSubscriptionTypes.SYSTEM:
             return None
         else:
             return NotImplementedError(
                 f"Unknown subscription type: {subscription_type}"
             )
 
+    def dispatch_unread_notifications_email(
+        self, *, site, unread_notification_count=None
+    ):
+        self.notification_email_last_sent_at = now()
+        self.save(update_fields=["notification_email_last_sent_at"])
+
+        send_unread_notifications_email(
+            site=site,
+            user=self.user,
+            n_notifications=(
+                unread_notification_count
+                if unread_notification_count
+                else self.unread_notification_count
+            ),
+        )
+
+    def dispatch_unread_direct_messages_email(
+        self, *, site, new_unread_message_count=None
+    ):
+        new_senders = [s.first_name for s in get_new_senders(user=self.user)]
```

Comment:
True, if they receive mutliple DMs at the same time, they would get multiple emails that each mention multiple senders, rather than just the one sender that the email was triggered by. I will add another argument to the function to fix that.